src/opusrtp.c

```diff
@@ -787,6 +787,72 @@ void write_packet(u_char *args, const struct pcap_pkthdr *header,
   }
 }
 
+int extract(const char* input_file)
+{
+  state *params;
+  pcap_t *pcap;
+  char errbuf[PCAP_ERRBUF_SIZE];
+  ogg_packet *op;
+
+  if ((pcap = pcap_open_offline(input_file, errbuf)) == NULL)
+  {
+    fprintf(stderr,"\nError opening dump file\n");
+    return -1;
+  }
+
+  params = malloc(sizeof(state));
+  if (!params) {
+    fprintf(stderr, "Couldn't allocate param struct.\n");
+    return -1;
```

Comment:
I've forgotten how the pcap api works. Should we call `pcap_close(pcap);` here and in all the subsequent cleanup clauses?